File: classes/apis/GoodReadApiCaller.php
<?php

/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
include_once 'ApiCaller.php';

/**
 * Description of GoodReadApiCaller
 *
 * @author Rabah
 */
class GoodReadApiCaller extends ApiCaller{

    //put your code here
    var $key = "IJ9e4JY4Gnl1JU5ADSYYg";
    var $url_search = "http://www.goodreads.com/search.xml";
    var $url_author = "http://www.goodreads.com/author/show/";

    public function searchAuthor($author) {
        $author_ = urlencode($author);
        $url_ = $this->url_search . "?key=" . $this->key . "&q=" . $author_ . "&search[field]=author";
        $result = $this->callApi($url_);
        //var_dump($result);
        //echo "<br/>";
        
    }

}

?>

File: websemantique/entities/src/main/java/um2/websemantique/entities/apicallers/GoodReadApiCaller.java
package um2.websemantique.entities.apicallers;

import um2.websemantique.entities.base.AuthorGoodRead;
import um2.websemantique.entities.utils.GeneratorFromXML;

public class GoodReadApiCaller extends ApiCaller {
	private String key = "IJ9e4JY4Gnl1JU5ADSYYg";
	private String urlSearch = "http://www.goodreads.com/search.xml";
	private String urlAuthor = "http://www.goodreads.com/author/show?id=";

	/**
	 * getting author xml Description from Good Read
	 * 
	 * @param author
	 * @return
	 */
	public AuthorGoodRead findGoodReadAuthor(String author) {
		author = ApiCaller.urlEncode(author);
		String url = urlSearch + "?key=" + key + "&q=" + author
				+ "&search[field]=author";
		String id = GeneratorFromXML.getIdGoodReadAuthor(url);

		url = urlAuthor + id + "&key=" + key;
		AuthorGoodRead result = GeneratorFromXML.getGoodReadAuthor(url);
		return result;
	}
}
